tutorials/parameter_sweep_demo.ipynb

```diff
@@ -0,0 +1,783 @@
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "num_samples = 4\n",
+    "num_procs = 4\n",
+    "ps, kwargs_dict = create_parameter_sweep_object(num_samples, num_procs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6f7ce19d",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "results_array, results_dict = ps.parameter_sweep(\n",
+    "    kwargs_dict[\"build_model\"],\n",
+    "    kwargs_dict[\"build_sweep_params\"],\n",
+    "    build_outputs = kwargs_dict[\"build_outputs\"],\n",
+    "    build_outputs_kwargs = kwargs_dict[\"build_outputs_kwargs\"],\n",
+    "    num_samples = num_samples,\n",
+    "    seed=None,\n",
+    "    build_model_kwargs = kwargs_dict[\"build_model_kwargs\"],\n",
+    "    build_sweep_params_kwargs = kwargs_dict[\"build_sweep_params_kwargs\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c648a402",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "pprint(results_dict)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5410d0c3",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "pprint(results_array)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7d14aba1",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Plotting Results\n",
+    "\n",
+    "The resulting H5 files can then be prcessed to generate plots, e.g., 2D scatter plots, map plots, etc.\n",
+    "\n",
+    "<table><tr>\n",
+    "    <td> <img src=\"assets_parameter_sweep_demo/scatter_LCOW_vs_Acomp.jpg\" alt=\"LCOW vs water permeability\" width=\"100%\"/> </td>\n",
+    "    <td> <img src=\"assets_parameter_sweep_demo/maps_LCOW_recovery_NaCl_loading.jpg\" alt=\"LCOW vs water recovery and NaCL concentration\" width=\"100%\"/> </td>\n",
+    "</tr></table>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d95eb8b4",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Advanced Usage and Features"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ea976e5",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Types of Parameter Sweeps\n",
+    "\n",
+    "* In its current form, a user can run 3 types of parameter sweeps\n",
+    "    - Simple parameter sweep\n",
+    "    - **Recursive parameter sweep**\n",
+    "    - Differential parameter sweep\n",
+    "* Samples for the parameter sweep can be generated from a probability distribution or an *n*-dimensional euclidean space where *n* is the number of parameters in the sweep. Latin hypercube sampling is also enabled."
```

Comment:
I think we should add the simplest (most concise) code block possible to demonstrate exactly how to call/run recursive parameter sweep.

Reply:
It should be there now.